### PR TITLE
[OSP] MachineSet primary subnet support

### PR DESCRIPTION
**Pull request by maxwelldb**
Context: https://issues.redhat.com/browse/OSDOCS-1663

### CODE REVIEW

**Comment by maxwelldb** [comment]
FYI @iamemilio

**Comment by maxwelldb** [comment]
Review URL is way wrong. TY, previewbot. 

* https://setting-osp-machine-subnet-osdocs1663--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-osp.html#machineset-yaml-osp_creating-machineset-osp

* https://setting-osp-machine-subnet-osdocs1663--ocpdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/installing-openstack-installer-custom.html#installation-osp-custom-subnet_installing-openstack-installer-custom

**Comment by iamemilio** [comment]
lgtm

**Comment by maxwelldb** [comment]
TY!

File: modules/installation-osp-custom-subnet.adoc [hunk 1]
```diff
@@ -11,22 +11,24 @@
...
-* DHCP is enabled on the target subnet.
+Clusters that use custom subnets have the following limitations:
 
-* You can provide installer credentials that have permission to create ports on the target network.
-
-* If your network configuration requires a router, it is created in {rh-openstack}. Some configurations rely on routers for floating IP address translation.
-
-* Your network configuration does not rely on a provider network. Provider networks are not supported.
+* If you plan to install a cluster that uses floating IPs, the `platform.openstack.machinesSubnet` subnet must be attached to a router that is connected to the `externalNetwork` network.
+* The installer cannot create a private network or subnet for your {rh-openstack} machines if the `platform.openstack.machinesSubnet` value is set in the `install-config.yaml`.
```
Comment (wjiangjay) [inline modules/installation-osp-custom-subnet.adoc:25]: `The installer cannot` or `The installer do not` ?

Comment (maxwelldb) [reply modules/installation-osp-custom-subnet.adoc:25]: My guess is that "does not" is more accurate. @iamemilio ?

Comment (maxwelldb) [inline modules/installation-osp-custom-subnet.adoc:25]:
```suggestion
* The installer does not create a private network or subnet for your {rh-openstack} machines if the `platform.openstack.machinesSubnet` value is set in the `install-config.yaml`.
```

Comment (wjiangjay) [reply modules/installation-osp-custom-subnet.adoc:25]: yeah, you are right, sorry for my poor english.

**Comment by wjiangjay** [comment]
LGTM

**Comment by maxwelldb** [comment]
Thanks!

**Comment by maxwelldb** [comment]
@openshift/team-documentation PTAL when you have a moment. Thanks!

File: modules/installation-osp-custom-subnet.adoc [hunk 2]
```diff
@@ -11,22 +11,23 @@
 
 Optionally, you can deploy a cluster on a {rh-openstack-first} subnet of your choice. The subnet's GUID is passed as the value of `platform.openstack.machinesSubnet` in the `install-config.yaml` file.
 
-This subnet is used as the cluster's primary subnet; nodes and ports are created on it.
+This subnet is used as the cluster's primary subnet. By default, nodes and ports are created on it. You can create nodes and ports on a different {rh-openstack} subnet by setting the value of the `platform.openstack.machinesSubnet` property to the subnet's UUID.
 
 Before you run the {product-title} installer with a custom subnet, verify that:
```
Comment (kalexand-rh) [inline modules/installation-osp-custom-subnet.adoc:16]: > Generally, use a sentence to introduce a list that is not a procedure.

File: modules/installation-osp-custom-subnet.adoc [hunk 3]
```diff
@@ -11,22 +11,23 @@
...
 
-* The target network and subnet are available.
+* The subnet that is used by `platform.openstack.machinesSubnet` has DHCP enabled.
+* The CIDR of `platform.openstack.machinesSubnet` matches the CIDR of `networking.machineNetwork`.
+* The installer user has permission to create ports on this network, including ports with fixed IP addresses.
```
Comment (kalexand-rh) [inline modules/installation-osp-custom-subnet.adoc:20]:
```suggestion
* The installation program user has permission to create ports on this network, including ports with fixed IP addresses.
```

File: modules/installation-osp-custom-subnet.adoc [hunk 4]
```diff
@@ -11,22 +11,23 @@
...
 
-* DHCP is enabled on the target subnet.
+Clusters that use custom subnets have the following limitations:
 
-* You can provide installer credentials that have permission to create ports on the target network.
-
-* If your network configuration requires a router, it is created in {rh-openstack}. Some configurations rely on routers for floating IP address translation.
-
-* Your network configuration does not rely on a provider network. Provider networks are not supported.
+* If you plan to install a cluster that uses floating IPs, the `platform.openstack.machinesSubnet` subnet must be attached to a router that is connected to the `externalNetwork` network.
```
Comment (kalexand-rh) [inline modules/installation-osp-custom-subnet.adoc:24]: > IP address noun
Use "IP address" or "Internet Protocol address", not "Internet address".
```suggestion
* If you plan to install a cluster that uses floating IP addresses, the `platform.openstack.machinesSubnet` subnet must be attached to a router that is connected to the `externalNetwork` network.
```

File: modules/installation-osp-custom-subnet.adoc [hunk 5]
```diff
@@ -11,22 +11,23 @@
...
+* The installer does not create a private network or subnet for your {rh-openstack} machines if the `platform.openstack.machinesSubnet` value is set in the `install-config.yaml`.
```
Comment (kalexand-rh) [inline modules/installation-osp-custom-subnet.adoc:25]:
```suggestion
* The installation program does not create a private network or subnet for your {rh-openstack} machines if the `platform.openstack.machinesSubnet` value is set in the `install-config.yaml` file.
```

> Generally, do not use a file name, file name extension, file type, or directory name as a noun.

**Review by kalexand-rh — COMMENTED** [review]
A few picky things, but it largely LGTM

File: modules/installation-osp-custom-subnet.adoc (hunk 2 above)
Comment (maxwelldb) [inline modules/installation-osp-custom-subnet.adoc:16]:
```suggestion
Before you run the {product-title} installer with a custom subnet, verify that your configuration meets the following requirements:
```

File: modules/installation-osp-custom-subnet.adoc (hunk 5 above)
Comment (maxwelldb) [inline modules/installation-osp-custom-subnet.adoc:25]:
```suggestion
[NOTE]
====
The installation program does not create a private network or subnet for your {rh-openstack} machines if the `platform.openstack.machinesSubnet` value is set in the `install-config.yaml` file.
====
```

Comment (maxwelldb) [reply modules/installation-osp-custom-subnet.adoc:25]: Resolving elsewhere.

**Comment by maxwelldb** [comment]
@kalexand-rh TY!

File: modules/installation-osp-custom-subnet.adoc [hunk 6]
```diff
@@ -11,19 +11,22 @@
...
-* DHCP is enabled on the target subnet.
+Clusters that use custom subnets have the following limitations:
 
-* You can provide installer credentials that have permission to create ports on the target network.
-
-* If your network configuration requires a router, it is created in {rh-openstack}. Some configurations rely on routers for floating IP address translation.
-
-* Your network configuration does not rely on a provider network. Provider networks are not supported.
+* If you plan to install a cluster that uses floating IP addresses, the `platform.openstack.machinesSubnet` subnet must be attached to a router that is connected to the `externalNetwork` network.
+[NOTE]
```
Comment (maxwelldb) [inline modules/installation-osp-custom-subnet.adoc:25]:
```suggestion
+
[NOTE]
```

**Pull request merged**